internal/friendbot.go

```diff
@@ -33,3 +35,9 @@ func (bot *Bot) Pay(ctx context.Context, destAddress string) (*TransactionResult
 	close(resultChan)
 	return maybeSubmitResult.maybeTransactionSuccess, maybeSubmitResult.maybeErr
 }
+
+// SupportsContractAddresses returns true if the bot is configured to fund
+// contract addresses (C addresses) and the network client supports it.
+func (bot *Bot) SupportsContractAddresses() bool {
+	return bot.FundContractAddresses && bot.NetworkClient.SupportsContractAddresses()
+}
```

Comment:
Is FundContractAddresses necessary config switch:
* Why would a friendbot instance need to disallow this feature other than if using horizon.
* It seems like retaining legacy horizon support is leaking into the main app and granular config. 

For consideration, can `FundContractAddresses` and `SupportsContractAddresses` be removed in favor of convention - 'the bot attempts to fund any address and returns an error at request time if it cannot'. The `minion.makeTx` request handler could be one place to validate at pay request time by calling `networkClient..GetAccountDetails(destAddress)` which provides err indication of non-contract support  `ErrContractAddressNotSupported`.